File: src/main/java/org/c3lang/intellij/psi/impl/C3LocalDeclarationStmtMixinImpl.java
package org.c3lang.intellij.psi.impl;

import com.intellij.lang.ASTNode;
import org.c3lang.intellij.psi.C3LocalDeclarationStmt;
import org.c3lang.intellij.psi.FullyQualifiedName;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public abstract class C3LocalDeclarationStmtMixinImpl extends C3PsiElementImpl implements C3LocalDeclarationStmt
{
	public C3LocalDeclarationStmtMixinImpl(@NotNull ASTNode node)
	{
		super(node);
	}

	@Override
	public @Nullable FullyQualifiedName findTypeName()
	{
		return FullyQualifiedName.from(getOptionalType());
	}
}

File: src/main/java/org/c3lang/intellij/psi/impl/C3LocalDeclarationStmtMixinImpl.kt
package org.c3lang.intellij.psi.impl

import com.intellij.lang.ASTNode
import org.c3lang.intellij.psi.C3LocalDeclarationStmt
import org.c3lang.intellij.psi.FullyQualifiedName

abstract class C3LocalDeclarationStmtMixinImpl(node: ASTNode) : C3LocalDeclarationStmt, C3PsiElementImpl(node) {

    override fun findTypeName(): FullyQualifiedName? {
        return FullyQualifiedName.from(optionalType)
    }
}